desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -161,11 +168,14 @@ def _make_unfurl_payload(url, id_type, doc, doc_type):
   request = MockRequest(user=user)
 
   if id_type == 'editor':
+    # Max rows allowed in result preview table
+    max_rows = 2
+
     try:
       status = _check_status(request, operation_id=doc_data['uuid'])
       if status['query_status']['status'] == 'available':
-        fetch_result = query_result(request, json.loads(doc.data))
-        unfurl_result = _make_result_table(fetch_result) if isinstance(fetch_result, dict) else fetch_result
+        fetch_result = query_result(request, json.loads(doc.data), max_rows)
+        unfurl_result = _make_result_table(fetch_result) if isinstance(fetch_result, list) else fetch_result
```

Comment:
`isinstance(fetch_result, list)` --> `if fetch_result`
?

actually, why do we care? (empty results is [])

Reply:
We are returning a string `Query result has expired or could not be found` whenever the status != available 
Because of this, we are checking is its a dict of result then sending it to table function else not @romainr

Reply:
Yes, this is bad practice.

Could return [] or None instead and handle the messaging there

Reply:
Oh okay